### ISSUE TITLE
this._env.run(...).on is not a function

### ISSUE THREAD

**Issue by amiramw** [camel-tooling/vscode-yeoman#44]
After selecting a generator from the quick pick there is an error notification: 
`this._env.run(...).on is not a function`

The flow continues successfully despite the error.

https://github.com/camel-tooling/vscode-yeoman/blob/035facb5c64acdb2f23e8bdcf0bab464a7aa439b/src/yo/yo.ts#L99-L115
There are several `on` calls on the expression evaluation of `this._env.run(generator, this.done)`. But the expression return a promise so on is not a function.

Should all the `on` calls be removed?

**Comment by bfitzpat** [comment]
Interesting. My intent for the extension was to get past the issues that weren't getting resolved on the root project, so we just cloned it so we could get past it. Many of these issues are likely due to the fact that the code is based on a much older (probably 2 years?) version of VS Code.

It would likely be beneficial to go through some of these types of issues and see what can be done to migrate the code forward before we simply remove it.

**Comment by amiramw** [comment]
it's hard to understand how it used to work few years ago but as far as I can see yeoman-generator still has these events but the generator instance is not exposed when calling environmnt.run. Only a promise to the completion of the process and a done callback are exposed.

**Comment by bfitzpat** [comment]
Is this._env.run a "thenable" by chance? Wondering if we can transition from "on" to "then"?

**Comment by amiramw** [comment]
Yes. It is thenable. So you mean we can replace the code above with something like this?

```
this._env.run(generator, this.done)
	.then(() => {
		this.clearState();
		console.log(`${EOL}${figures.tick} done`);
	});
```

(the bower and npm event handlers will be removed)

**Comment by amiramw** [comment]
Also catch error can be implemented. I'll try to update the PR.

**Comment by amiramw** [comment]
Updated PR #49. Please review.

**Comment by bfitzpat** [comment]
I've put a comment in the PR, but have a big concern about losing any functionality since we don't have any tests really to ensure that we're not breaking something unexpectedly. It's one of the things the initial project was missing. I added a very minimal test or two when I forked, but that's it. 

Perhaps we should create an issue to handle testing separately - but I think it deserves some attention.
